onnxruntime/core/framework/bfc_arena.cc

```diff
@@ -82,6 +84,12 @@ bool BFCArena::Extend(size_t rounded_bytes) {
     } catch (const std::bad_alloc&) {
       // attempted allocation can throw std::bad_alloc. we want to treat this the same as if it returned nullptr
       // so swallow the exception
+    } catch (const OnnxRuntimeException& ort_exception) {
+      // swallow if exception is our throw from a failed cudaMalloc call.
+      // re-throw otherwise.
+      if (std::string(ort_exception.what()).find("cudaMalloc") == std::string::npos) {
```

Comment:
Is there value in logging a message (cause of the exception, etc) when swallowing the exception?

Reply:
I didn't add logging given it's not fatal at this point and the log messages could be confusing noise. If all the backoff iterations fail the change on line 114 will result in an error message which is logged via the usual exception handling.